tasks/relations.js

```diff
@@ -0,0 +1,72 @@
+import Listr from "listr";
+import { apiV8, apiV9 } from "../api.js";
+import { writeContext } from "../index.js";
+
+export async function migrateRelations(context) {
+  return new Listr([
+    {
+      title: "Migrating Relations",
+      task: () => migrateRelationsData(context),
+    },
+    {
+      title: "Saving Relations context",
+      task: () => writeContext(context, "relations"),
+    },
+  ]);
+}
+
+async function migrateRelationsData(context) {
+  const relations = await apiV8.get("/relations", { params: { limit: -1 } });
+  const relationsV9 = relations.data.data
+    .filter((relation) => {
+      return (
+        (relation.collection_many.startsWith("directus_") &&
+          relation.collection_one.startsWith("directus_")) === false
+      );
+    })
+    .map((relation) => ({
+      // @NOTE: one_primary will be removed from Directus soon, so i'm not too worried about it here
+      meta: {
+        many_collection: relation.collection_many,
+        many_field: relation.field_many,
+        many_primary: "id",
+        one_collection: relation.collection_one,
+        one_field: relation.field_one,
+        one_primary: "id",
+        junction_field: relation.junction_field,
+      },
```

Comment:
many_primary and one_primary have indeed since been removed

Reply:
Done